src/clocksi_interactive_tx_coord_fsm.erl

```diff
@@ -362,18 +345,55 @@ execute_op({OpType, Args}, Sender,
                 ReadResult ->
                     {reply, {ok, ReadResult}, execute_op, SD0}
             end;
-        update ->
-            case perform_update(Args, Updated_partitions, Transaction, Sender, ClientOps) of
-                {error, _Reason} ->
-                    abort(SD0#tx_coord_state{from = Sender});
-                {NewUpdatedPartitions, NewClientOps} ->
-                    {next_state, execute_op,
-                        SD0#tx_coord_state{updated_partitions = NewUpdatedPartitions,
-                                          client_ops = NewClientOps}}
-            end
+	    update_objects ->
+		    ExecuteUpdates =
+                fun({Key, Type, UpdateParams}, Acc = #tx_coord_state{updated_partitions=UpdatedPartitions, client_ops=ClientOps}) ->
+                    case perform_update({Key, Type, UpdateParams}, UpdatedPartitions, Transaction, Sender, ClientOps) of
+                        {error, Reason} ->
+                            Acc#tx_coord_state{return_accumulator= {error, Reason}};
+                        {NewUpdatedPartitions, NewClientOps} ->
+                            NewNumToRead = Acc#tx_coord_state.num_to_read,
+                            Acc#tx_coord_state{num_to_read =  NewNumToRead+1,
+                                updated_partitions=NewUpdatedPartitions, client_ops=NewClientOps}
+			        end
+		    end,
+		    NewCoordState = lists:foldl(ExecuteUpdates, SD0#tx_coord_state{num_to_read = 0, return_accumulator= ok}, Args),
+		    case NewCoordState#tx_coord_state.num_to_read > 0 of
+			    true ->
+				    {next_state, receive_logging_responses, NewCoordState#tx_coord_state{from = Sender}};
+			    false ->
+				    {next_state, receive_logging_responses, NewCoordState#tx_coord_state{from = Sender}, 0}
+		    end
     end.
 
 
+%% @doc This state reached after an execute_op(update_objects[Params]).
+%% update_objects calls the perform_update function, which asynchronously
+%% sends a log operation per update, to the vnode responsible of the updated
+%% key. After sending all those messages, the coordinator reaches this state
+%% to receive the responses of the vnodes.
+receive_logging_responses(Response, S0 = #tx_coord_state{num_to_read = NumToReply,
+	                        return_accumulator= ReturnAcc}) ->
+	NewAcc = case Response of
+		{error, Reason} -> {error, Reason};
+		{ok, _OpId} -> ReturnAcc;
+		timeout -> ReturnAcc
+	end,
+	case NumToReply > 1 of
+		false ->
+			gen_fsm:reply(S0#tx_coord_state.from, NewAcc),
```

Comment:
It should be in "true" of the following case statement. Otherwise if it is an error, it replies a message here and then initiates abort where it sends a reply again.